src/data/repositories/UserD2Repository.ts

```diff
@@ -2,26 +2,59 @@ import { User } from "$/domain/entities/User";
 import { UserRepository } from "$/domain/repositories/UserRepository";
 import { D2Api, MetadataPick } from "$/types/d2-api";
 import { apiToFuture, FutureData } from "$/data/api-futures";
+import { array, Codec, string } from "purify-ts";
+import { Future } from "$/domain/entities/generic/Future";
+import { Id } from "$/domain/entities/Ref";
+import _c from "$/domain/entities/generic/Collection";
 
 export class UserD2Repository implements UserRepository {
     constructor(private api: D2Api) {}
 
     public getCurrent(): FutureData<User> {
-        return apiToFuture(
+        const administratorGroups$ = apiToFuture(
+            this.api.models.constants.get({
+                fields: constantFields,
+                filter: { code: { eq: "TALLY_SHEETS_STORAGE" } },
+                paging: false,
+            })
+        ).flatMap(res => {
+            const description = _c(res.objects).first()?.description;
+
+            /* To pass beside not having the constant or not being valid */
+            if (description) {
+                return constantDescriptionCodec.decode(JSON.parse(description)).caseOf({
+                    Left: _err => {
+                        console.error(new Error(constantsErrMsg));
+                        return Future.success<Error, Id[]>([]);
+                    },
+                    Right: res => Future.success<Error, Id[]>(res.administratorGroups),
+                });
+            } else {
+                console.error(new Error(constantsErrMsg));
+                return Future.success<Error, Id[]>([]);
+            }
+        });
+
+        const d2User$ = apiToFuture(
             this.api.currentUser.get({
                 fields: userFields,
             })
-        ).map(d2User => {
-            const res = this.buildUser(d2User);
-            return res;
-        });
+        );
+
+        return d2User$.chain(d2User =>
+            administratorGroups$.map(adminGroups => this.buildUser(d2User, adminGroups))
+        );
     }
 
-    private buildUser(d2User: D2User) {
+    private buildUser(d2User: D2User, adminGroups: string[]) {
+        const canSelectAllLocales = d2User.userGroups.some(group => adminGroups.includes(group.id));
```

Comment:
[declarative] Very minor, but in my experience it works better to name the variable not describing how it's going to be used, but what conceptually what the value holds. Example: `userBelongsToSomeAdminGroup`. An then, when used, it can be read as documentation: 

```
            canSelectAllLocales: userBelongsToSomeAdminGroup,
```